File: DWES_desarrollo_web_entorno_servidor/ejercicios_libro/ejercicios12_7/ejercicio3/Cliente.php
<?php
require_once 'TiendaEjercicio3__12_7.php';

class Cliente
{
    // ATRIBUTOS
    private $nombre;
    private $token;
    private $peticiones;

    // CONSTRUCTOR
    public function __construct(
        $nombre = "",
        $token = "",
        $peticiones = 0
    ) {
        $this->nombre = $nombre;
        $this->token = $token;
        $this->peticiones = $peticiones;
    }

    // GETTERS
    public function getNombre()
    {
        return $this->nombre;
    }

    public function getToken()
    {
        return $this->token;
    }

    public function getPeticiones()
    {
        return $this->peticiones;
    }

    // SETTERS
    public function setPeticiones($cantidad)
    {
        $this->peticiones = $cantidad;
    }

    // METODOS
    public static function validarToken($token)
    {
        $conexion = TiendaEjercicio3__12_7::connectDB();
        $consulta = $conexion->query("SELECT * FROM cliente WHERE token = '$token'");

        if ($consulta->rowCount() > 0) {
            return true;
        }

        return false;
    }

    public static function sumarPeticion($token)
    {
        $conexion = TiendaEjercicio3__12_7::connectDB();
        $consulta = $conexion->query("UPDATE cliente SET peticiones = peticiones + 1 WHERE token = '$token'");
    }
}

File: HLC_hora_de_libre_configuracion/proyectos/Biblioteca/Cliente.java
package HLC.trimestre1.Proyecto.Biblioteca;

public class Cliente {
    // ATRIBUTOS
    protected String dni;
    protected String nombre;

    // CONSTRUCTOR
    public Cliente(String dni, String nombre) {
        this.dni = dni;
        this.nombre = nombre;
    }

    // GETTERS AND SETTERS
    public String getDni() {
        return dni;
    }

    public void setDni(String dni) {
        this.dni = dni;
    }

    public String getNombre() {
        return nombre;
    }

    public void setNombre(String nombre) {
        this.nombre = nombre;
    }

    // METODOS
    @Override
    public String toString() {
        return "Nombre: " + this.nombre + "\nDNI: " + this.dni + "\n";
    }
}
